### PR TITLE
ESB-138 LessonController to Automapper and fix some bug

### PR DESCRIPTION
**Pull request by valeevden**

### CODE REVIEW

File: EducationSystem/EducationSystem.DB/StoredProcedures/Course/dbo.Course_SelectAll.sql [hunk 1]
```diff
@@ -10,5 +10,5 @@ begin
 		t.Name
 	from dbo.Course c left join dbo.Course_Theme ct on ct.CourseID = c.Id
 		left join dbo.Theme t on   ct.ThemeID=t.Id
-	where IsDeleted=0
+	where c.IsDeleted=0
 end
```
Comment (Negodyaj) [inline EducationSystem/EducationSystem.DB/StoredProcedures/Course/dbo.Course_SelectAll.sql:14]: could you please recreate dbo.Lesson_SelectByID.sql (which is shown as BIN right after this file)?
and please remove l.IsDeleted=0 from "where" part of dbo.Lesson_SelectByID

File: EducationSystem/EducationSystem/Controllers/LessonController.cs [hunk 2]
```diff
@@ -21,14 +21,14 @@ namespace EducationSystem.Controllers
     public class LessonController : ControllerBase
     {
         private ILessonRepository _repo;
-        private LessonMapper _lessonMapper;
+        //private LessonMapper _lessonMapper;
```
Comment (Negodyaj) [inline EducationSystem/EducationSystem/Controllers/LessonController.cs:24]: please remove such comments

File: EducationSystem/EducationSystem/Controllers/LessonController.cs [hunk 3]
```diff
@@ -41,7 +41,7 @@ public ActionResult AddNewLesson([FromBody] LessonInputModel inputModel)
             LessonDto lesson;
             try
             {
-                lesson = _lessonMapper.ToDto(inputModel);
+                lesson =_mapper.Map<LessonDto>(inputModel);
```
Comment (Negodyaj) [inline EducationSystem/EducationSystem/Controllers/LessonController.cs:44]: the registration of "CreateMap<LessonInputModel,LessonDto>();" requires some custom mapping:
int GroupID => GroupDto Group
string LessonDate => DateTime LessonDate
List<int> ThemesId => List<ThemeDto> Themes

File: EducationSystem/EducationSystem/Controllers/LessonController.cs [hunk 4]
```diff
@@ -41,7 +41,7 @@ public ActionResult AddNewLesson([FromBody] LessonInputModel inputModel)
             LessonDto lesson;
             try
             {
-                lesson = _lessonMapper.ToDto(inputModel);
+                lesson =_mapper.Map<LessonDto>(inputModel);
             }
             catch (Exception ex)
```
Comment (Negodyaj) [inline EducationSystem/EducationSystem/Controllers/LessonController.cs:46]: a huge part of methods below uses dto as input parameter, and it must be changed to input models with further mapping
the same thin with return types: we are getting dtos from services, and then we must map it into output models
and, of course, direct using of repositories' methods is restricted. all this stuff must be changed to call the business layer

**Comment by valeevden** [comment]
Я все переделал и создал новый пул-реквест #29

**Pull request closed without merging**